### DISCUSSION TITLE
applying delineate_crowns() to lascatalog

### DISCUSSION THREAD

**Discussion by albatty** [r-lidar/lidR#592] [Q&A]
I have successfully applied dalponte2016 to a LAScatalog file to segment trees. I am now trying to extract the crowns as polygons and export them as a shape file. However delineate_crowns() does not work with LAScatalog, only on a las/laz file. Is there an alternative to extract the polygons?

Cheers:)
Robin

**Comment by Jean-Romain** [comment]
Which version of the package are you using? What does _"it does not works"_ means?

**Comment by albatty** [comment]
I'm using 4.0.1. I am getting the error:

```r
Error in lazyeval::f_capture(func) : Promise has already been forced
```

In the documentation it states `delineate_crowns()` takes and 'object of class LAS*'.

**Reply by Jean-Romain** [reply]
Please show your code. Is `func` and expression? The doc states `LAS*` which means `LAS`, `LAScatalog` or `LAScluster`. It also states that this function is deprecated. Yet it should works.

**Reply by albatty** [reply]
performing the segmentation

```r
opt_output_files(ctg_norm) <- paste0(tempdir(), "/{*}_segmented")
dalponte <- dalponte2016(chm, th_tree = 4, ttops, max_cr = 25)
ctg_segmented <- segment_trees(ctg_norm, dalponte)
```
this is the line that is giving the error message
```r
dalponte_crowns <- delineate_crowns(ctg_segmented) %>% st_as_sf(dalponte_crowns)
```

**Reply by Jean-Romain** [reply]
You found a bug

**Reply by Jean-Romain** [reply]
See #594 . As a temporary workaround you can compute a dummy metric

```r
dalponte_crowns <- crown_metrics(ctg_segmented, func = ~list(0)) 
```
